botorch/optim/optimize_mixed.py

```diff
@@ -815,14 +885,31 @@ def optimize_acqf_mixed_alternating(
         A tuple of two tensors: a (q x d)-dim tensor of optimized points
             and a (q)-dim tensor of their respective acquisition values.
     """
+
     if sequential is False:  # pragma: no cover
         raise NotImplementedError(
             "`optimize_acqf_mixed_alternating` only supports "
             "sequential optimization."
         )
 
     cat_dims = cat_dims or []
-    discrete_dims = discrete_dims or []
+    discrete_dims = discrete_dims or {}
+
+    discrete_dims = OrderedDict(sorted(discrete_dims.items(), key=lambda x: x[0]))
+    # sort the values in discrete dims in ascending order
+    discrete_dims = {dim: sorted(values) for dim, values in discrete_dims.items()}
+
+    for dim, values in discrete_dims.items():
+        if not float(values[0]) == bounds[0, dim].item():
+            raise ValueError(
+                f"Discrete dimension {dim} must start at the lower bound "
+                f"{bounds[0, dim].item()} but starts at {values[0]}."
+            )
+        if not float(values[-1]) == bounds[1, dim].item():
+            raise ValueError(
+                f"Discrete dimension {dim} must end at the upper bound "
+                f"{bounds[1, dim].item()} but end at {values[-1]}."
+            )
```

Comment:
```suggestion
    for dim, values in discrete_dims.items():
        lower_bnd, upper_bnd = bounds[:, dim].tolist()
        lower, upper = values[0], values[-1]
        if lower != lower_bnd:
            raise ValueError(
                f"Discrete dimension {dim} must start at the lower bound "
                f"{lower_bnd} but starts at {lower}."
            )
        if upper != upper_bnd:
            raise ValueError(
                f"Discrete dimension {dim} must end at the upper bound "
                f"{upper_bnd} but end at {upper}."
            )
```

Reply:
Thanks for the hint, I updated accordingly.